src/freenet/client/filter/ContentFilter.java

```diff
@@ -368,6 +361,16 @@ public static FilterStatus filter(InputStream input, OutputStream output, String
 			}
 
 			if(handler.safeToRead) {
+				if(handler.takesACharset && ((charset == null) || (charset.isEmpty()))) {
+					byte[] charsetBuffer = new byte[CHARSET_DETECTION_FALLBACK_BUFFERSIZE];
+					int offset = readIntoBuffer(input, CHARSET_DETECTION_FALLBACK_BUFFERSIZE, charsetBuffer);
+					BOMDetection bom = CSSReadFilter.detectCharsetFromBOM(charsetBuffer, CHARSET_DETECTION_FALLBACK_BUFFERSIZE);
```

Comment:
I’m pretty sure this is 100% wrong. That method detects an encoding from the representation of the string `@charset`. It is also gloriously misnamed as it has nothing to do with a BOM. 😀